crates/core/src/logstore/default_logstore.rs

```diff
@@ -1,12 +1,25 @@
 //! Default implementation of [`LogStore`] for storage backends with atomic put-if-absent operation
 
-use std::sync::Arc;
+use std::sync::{Arc, OnceLock};
 
 use bytes::Bytes;
-use object_store::{path::Path, ObjectStore};
+use object_store::{Attributes, Error as ObjectStoreError, ObjectStore, PutOptions, TagSet};
 
-use super::{LogStore, LogStoreConfig};
-use crate::{operations::transaction::TransactionError, storage::ObjectStoreRef, DeltaResult};
+use super::{CommitOrBytes, LogStore, LogStoreConfig};
+use crate::{
+    operations::transaction::TransactionError,
+    storage::{commit_uri_from_version, ObjectStoreRef},
+    DeltaResult,
+};
+
+fn put_options() -> &'static PutOptions {
+    static PUT_OPTS: OnceLock<PutOptions> = OnceLock::new();
```

Comment:
why put it in a static lock?

Reply:
Was thinking because it needs to be called at every put call, but I guess it doesn't matter since I have to clone anyways

Reply:
I don't have any problems with this code, but it does increase the MRSV, so keep that in mind since OnceLock was added to std in 1.80 I believe

Reply:
@hntd187 we already use OnceLock in a couple other places ^^